src/core/uint.js

```diff
@@ -166,7 +186,7 @@ UInt.prototype.parse_generic = function(j) {
 };
 
 UInt.prototype.parse_hex = function(j) {
-  if (typeof j === 'string' && j.length === (this.constructor.width * 2)) {
+  if (new RegExp(`^[0-9A-F]{${this.constructor.width * 2}}$`).test(j)) {
```

Comment:
Shouldn't hex also include lower case `a-f` ? That would be a reasonable expectation?